### PR TITLE
[ml] Update command reference documentation

### PR DESCRIPTION
**Pull request by diondrapeck**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/parameterized_spark.py [hunk 1]
```diff
@@ -16,23 +16,24 @@
...
     """Spark component that contains supporting parameters.
 
-    :param code: The source code to run the job.
+    :param code: The source code to run the job. Can be a local path or "http:", "https:", or "azureml:" url pointing
+        to a remote location.
     :type code: str
-    :param entry: Entry.
-    :type entry: str
-    :param py_files: List of .zip, .egg or .py files to place on the PYTHONPATH for Python apps.
+    :param entry: The file or class entry point.
```
Comment (MilesHolland) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/parameterized_spark.py:22]: What is an entry point in this context? Also, why is the type changing dramatically from a string to a string dict? Based on the name I would've expected it to remain a string.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_component/command_component.py [hunk 2]
```diff
@@ -151,15 +173,19 @@ def _get_origin_code_value(self) -> Union[str, os.PathLike, None]:
...
+        :rtype: int
         """
         return self.resources.instance_count if self.resources else None
 
     @instance_count.setter
-    def instance_count(self, value: int):
+    def instance_count(self, value: int) -> None:
+        """Sets the number of instances or nodes to be used by the compute target.
+
+        :param value: Tne number of instances of nodes to be used by the compute target. Defaults to 1.
```
Comment (MilesHolland) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_component/command_component.py:186]: Nit Typo: Tne
```suggestion
        :param value: The number of instances of nodes to be used by the compute target. Defaults to 1.
```

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_component/command_component.py [hunk 3]
```diff
@@ -33,43 +33,65 @@
...
-    :param description: Description of the component.
+    :param description: The description of the component.
     :type description: str
     :param tags: Tag dictionary. Tags can be added, removed, and updated.
     :type tags: dict
-    :param display_name: Display name of the component.
+    :param display_name: The display name of the component.
     :type display_name: str
-    :param command: Command to be executed in component.
+    :param command: The command to be executed during training.
```
Comment (MilesHolland) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_component/command_component.py:48]: Why was the wording changed to 'during training'? I thought command components were more general purpose than that.

Comment (diondrapeck) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_component/command_component.py:48]: That was my perception as well before beginning this work, but the existing documentation (other than this particular description) regularly aligns Command Job with training. For example, look below at the description for the distribution parameter ("Distribution configuration for distributed training") and these existing documentation descriptions for Command-related entities which all call out training.

- "command", "environment", and "identity" descriptions for Command [here](https://learn.microsoft.com/en-us/python/api/azure-ai-ml/azure.ai.ml.entities.command?view=azure-python) and CommandJob [here](https://learn.microsoft.com/en-us/python/api/azure-ai-ml/azure.ai.ml.entities.commandjob?view=azure-python)

I'm open to removing those references if you think it's more representative of what can be done with a Command. We do control Command-related entities now, so it's within our description. What do you think?

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_component/command_component.py (hunk 2 above)
Comment (diondrapeck) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_component/command_component.py:186]: Unsettling that cspell didn't fail on this. Good catch!

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/parameterized_spark.py (hunk 1 above)
Comment (diondrapeck) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/parameterized_spark.py:22]: - Spark components are configured to make a Spark container inside which the user can install their files and run their code. The entry point is the industry keyword for a script executable to run when the container starts that will let you run the container as if it were that binary.
 - I updated it to match the type in the actual constructor. Seems like whoever initially wrote the docstring didn't confirm it with the actual types or the types were updated after without updating the docstring.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_component/command_component.py (hunk 3 above)
Comment (diondrapeck) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_component/command_component.py]: Discussed offline - training references have been removed in favor of generic language

**Review by MilesHolland — APPROVED** [review]

**Review by singankit — APPROVED** [review]

**Review by luigiw — APPROVED** [review]

**Pull request merged**